src/SFML/Window/Unix/WindowImplX11.cpp

```diff
@@ -825,6 +825,12 @@ Vector2u WindowImplX11::getSize() const
 ////////////////////////////////////////////////////////////
 void WindowImplX11::setSize(const Vector2u& size)
 {
+    // We could set the min and max to the new size in its respective dimension if setSize violates 
+    // previously set min and max sizes
+    assert(size.x > m_minimumSize.x);
+    assert(size.y > m_minimumSize.y);
+    assert(size.x < m_maximumSize.x);
+    assert(size.y < m_maximumSize.y);
```

Comment:
```suggestion
    assert(size.x >= m_minimumSize.x);
    assert(size.y >= m_minimumSize.y);
    assert(size.x <= m_maximumSize.x);
    assert(size.y <= m_maximumSize.y);
```

Also, do we really want to `assert` this? The user could call `setSize` from the public interface of `sf::Window`.

If we want to `assert` and not return a `bool`, we should add something like: *"The behavior is undefined if this function is called with a `size` that does not fit in the previously specified minimum and maximum size boundaries. If such boundaries have never been previously specified, any value of `size` is valid."* to the documentation of `Window::setSize`.

Reply:
This is lacking implementation. I do not think we should assert either, but we need to define what we want here. There are a few options:

1. Ignore the setSize call
2. Ignore the setSize call for the respective dimension effectively allowing the set size to work if either the new width or height is valid but only for the respective dimension.
3. of course we could still assert and crash the program
4. set a new minimum and maximum size in each respective dimension if the new setSize function violates the current minimum / maximum size

I think 4 is the right call, but do let me know! And it shouldn't be `<=`. We should allow users to perfectly butt up against the limits if they so choose to.

Reply:
@vittorioromeo Anyways, what about the other options i detailed here https://github.com/SFML/SFML/pull/2519#discussion_r1165808282